cmd/unused/internal/ui/csv_export.go

```diff
@@ -0,0 +1,70 @@
+package ui
+
+import (
+	"context"
+	"encoding/csv"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/grafana/unused"
+	"github.com/grafana/unused/cmd/internal"
+)
+
+func CsvExport(ctx context.Context, options Options) error {
+	disks, err := listUnusedDisks(ctx, options.Providers)
+	if err != nil {
+		return err
+	}
+
+	if options.Filter.Key != "" {
+		filtered := make(unused.Disks, 0, len(disks))
+		for _, d := range disks {
+			if d.Meta().Matches(options.Filter.Key, options.Filter.Value) {
+				filtered = append(filtered, d)
+			}
+		}
+		disks = filtered
+	}
+
+	if len(disks) == 0 {
+		return nil
+	}
+
+	w := csv.NewWriter(os.Stdout)
+	defer w.Flush()
```

Comment:
As in the case of the table writer we need to check the error returned by `Flush`, so perhaps we should not `defer` this and add an error check after the loop.